File: fizzbuzz/src/FizzBuzz.java

public class FizzBuzz {

	/*
	 
	public Object fizzbuzz(int i){
		Fizz fizz=new Fizz();
		Buzz buzz=new Buzz();
		Fb fb=new Fb();
		Object[] cadena=new Object[100];
		if(fizz.fizz(i)) cadena[1]="fizz";
		if(buzz.buzz(i)) cadena[1]="buzz";
		if(fb.fb(i)) cadena[1]="fizz buzz";
		if(cadena[1]!=""){
				return cadena;
		}
		return Integer.toString(i);
		
	}*/
	
	public String fizzbuzz(int i){
		Fizz fizz=new Fizz();
		Buzz buzz=new Buzz();
		Fb fb=new Fb();
		String cadena="";
		if(fizz.validate(i)) cadena="fizz";
		if(buzz.validate(i)) cadena="buzz";
		if(fb.validate(i)) cadena="fizz buzz";
		if(cadena!=""){
				return cadena;
		}
		return Integer.toString(i);
		
	}
	 
	public String print(int k){
		String result="";
		for(int i=1;i<=k;i++){
			if(i==1){
				result=fizzbuzz(i);
			}
			else result=result+", "+fizzbuzz(i);
		}
		return result;
	}
	

}

File: integracion/fizzBuzzArkaitz/app/scripts/FizzBuzz.js
// APP.FizzBuzz = (function(){})();
define('FizzBuzz', ['Fizz', 'Buzz'], function() {
    console.log('Fizz Buzz app');

    var validators = [];

    var testNumber = function(num){
        var n = num || 1,
            res = [];

        for (var i = 0; i < validators.length; i++) {
            if(validators[i].validate(n)) {
                res.push(validators[i].getValue());
            }
        }

        if(!res.length) {
            res.push(n);
        }

        return res.join('');
    };

    var print = function(num, join){
        var n = num || 1,
            glue = join|| ',',
            res = [];

        for (var i = 1; i <= n; i++) {
            res.push(testNumber(i));
        }

        return res.join(glue);
    };

    var addValidator = function(validator) {
        // TODO: Check if is instance of Validator
        if(validator.hasOwnProperty('validate')) {
            validators.push(validator);
        }
    };

    for (var i = 0; i < arguments.length; i++) {
        addValidator(arguments[i]);
    }

    return {
        testNumber : testNumber,
        print : print,
        addValidator : addValidator
    };
});